src/stan/mcmc/hmc/hamiltonians/base_hamiltonian.hpp

```diff
@@ -36,6 +36,8 @@ namespace stan {
         return T(z) + V(z);
       }
 
+      virtual double dG_dt(Point& z) = 0;
```

Comment:
Can this be `const Point&`?  It's much easier to get the const correctness in from the get go.  And same question for `dtau_dq()` below.